src/librustc_data_structures/array_vec.rs

```diff
@@ -81,15 +129,23 @@ impl<A: Array> Drop for ArrayVec<A> {
 impl<A: Array> Extend<A::Element> for ArrayVec<A> {
     fn extend<I>(&mut self, iter: I) where I: IntoIterator<Item=A::Element> {
         for el in iter {
-            unsafe {
-                let arr = &mut self.values as &mut [ManuallyDrop<_>];
-                arr[self.count].value = el;
-            }
-            self.count += 1;
+            self.push(el);
         }
     }
 }
 
+impl<A: Array> Iterator for ArrayVec<A> {
```

Comment:
Why? You could make a *forward* by-value iterator if you wanted to (by having an `ArrayVec` and a start index), and then implement `IntoIterator` on `ArrayVec`.